tests/commonTemplates_test.go

```diff
@@ -0,0 +1,155 @@
+			}, testTemplate.NewResource())
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("resource deletion", func() {
+		table.DescribeTable("recreate after delete", func(res *testResource) {
+			resource := res.NewResource()
+			resource.SetName(res.Name)
+			resource.SetNamespace(res.Namsespace)
+			Expect(apiClient.Delete(ctx, resource)).ToNot(HaveOccurred())
+
+			Eventually(func() error {
+				return apiClient.Get(ctx, client.ObjectKey{
+					Name: res.Name, Namespace: res.Namsespace,
+				}, resource)
+			}, timeout, time.Second).ShouldNot(HaveOccurred())
+		},
+			table.Entry("view role", viewRole),
+			table.Entry("view role binding", viewRoleBinding),
+			table.Entry("testTemplate in custom NS", testTemplate),
+		)
+		table.DescribeTable("recreate after delete", func(res *testResource) {
+			resource := res.NewResource()
+			resource.SetName(res.Name)
+			resource.SetNamespace(res.Namsespace)
+			Expect(apiClient.Delete(ctx, resource)).ToNot(HaveOccurred())
+
+			Eventually(func() error {
+				return apiClient.Get(ctx, client.ObjectKey{
+					Name: res.Name,
+				}, resource)
+			}, timeout, time.Second).ShouldNot(HaveOccurred())
+		},
+			table.Entry("edit cluster role", editClusterRole),
+			table.Entry("golden image NS", goldenImageNS),
+		)
+	})
+
+	Context("resource change", func() {
+		table.DescribeTable("should restore modified resource", func(
+			res *testResource,
+			updateFunc interface{},
+			equalsFunc interface{},
+		) {
+			key := res.GetKey()
+			original := res.NewResource()
+			Expect(apiClient.Get(ctx, key, original)).ToNot(HaveOccurred())
+
+			changed := original.DeepCopyObject()
+			reflect.ValueOf(updateFunc).Call([]reflect.Value{reflect.ValueOf(changed)})
+			Expect(apiClient.Update(ctx, changed)).ToNot(HaveOccurred())
+
+			newRes := res.NewResource()
+			Eventually(func() bool {
+				Expect(apiClient.Get(ctx, key, newRes)).ToNot(HaveOccurred())
+				res := reflect.ValueOf(equalsFunc).Call([]reflect.Value{
+					reflect.ValueOf(original),
+					reflect.ValueOf(newRes),
+				})
+				return res[0].Interface().(bool)
+			}, timeout, time.Second).Should(BeTrue())
+		},
+			table.Entry("edit cluster role", editClusterRole,
+				func(role *rbac.ClusterRole) {
+					role.Rules[0].Verbs = []string{"watch"}
+				},
+				func(old *rbac.ClusterRole, new *rbac.ClusterRole) bool {
+					return reflect.DeepEqual(old.Rules, new.Rules)
+				}),
+
+			table.Entry("view role", viewRole,
+				func(roleBinding *rbac.Role) {
+					roleBinding.Rules = []rbac.PolicyRule{}
+				},
+				func(old *rbac.Role, new *rbac.Role) bool {
+					return reflect.DeepEqual(old.Rules, new.Rules)
+				}),
+
+			table.Entry("view role binding", viewRoleBinding,
+				func(roleBinding *rbac.RoleBinding) {
+					roleBinding.Subjects = []rbac.Subject{}
+				},
+				func(old *rbac.RoleBinding, new *rbac.RoleBinding) bool {
+					return reflect.DeepEqual(old.Subjects, new.Subjects)
+				}),
+		)
```

Comment:
Can you add a `table.Entry` to test modification of a template?

Reply:
Added